### PR TITLE
Implement initial evm workspace, fix small bugs

### PR DESCRIPTION
**Pull request by mossberg**
- Fixes a bug in tuple size in sol compile helper
- generate .tx.# files for tx in each state
- save the compiled bytecode in workspace as a convenience, if solidity was compiled.

### CODE REVIEW

File: manticore/core/plugin.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Tracer(Plugin):
     def will_start_run_callback(self, state):
         state.context['trace'] = []
 
-    def did_execute_instruction_callback(self, state, pc, target_pc, instruction):
+    def will_execute_instruction_callback(self, state, pc, instruction):
```
Comment (feliam) [inline manticore/core/plugin.py]: Does this works on forking instructions ? Does this get executed twice on a fork ?

File: manticore/platforms/evm.py [hunk 2]
```diff
@@ -2381,3 +2382,27 @@ def compare_buffers(a, b):
             
         self.current._push(value)
         self.current.pc += self.current.instruction.size
+
+    def generate_workspace_files(self):
+        def solve_data(data):
+            ret = ''
+            try:
```
Comment (feliam) [inline manticore/platforms/evm.py]: Why not state.solve_one   ?

Comment (mossberg) [reply manticore/platforms/evm.py]: there is no state accessible from the platform

Comment (feliam) [reply manticore/platforms/evm.py]: Ah.
So why not 
`solver.get_value(self._constraints, data)`
?

Comment (mossberg) [reply manticore/platforms/evm.py]: ah, there is no reason, i wasn't paying attention :( just quickly copying `SLinux.generate_workspace_file`

File: manticore/core/plugin.py (hunk 1 above)
Comment (mossberg) [reply manticore/core/plugin.py]: yes it probably will. i don't have time to really think too much about a proper solution for this right now, and it's not that high priority, so i'll omit this diff

Comment (mossberg) [reply manticore/core/plugin.py]: good catch, thanks!

File: manticore/platforms/evm.py [hunk 3]
```diff
@@ -2381,3 +2382,18 @@ def compare_buffers(a, b):
...
+
+    def generate_workspace_files(self):
+        ret = {}
+        for i, tx in enumerate(self.transactions):
+            name = 'tx.{}'.format(i)
+            data = {
+                'to': tx.address,
+                'from': tx.caller,
+                'data': solver.get_value(self.constraints, tx.data).encode('hex'),
+                'value': tx.value if not issymbolic(tx.value) else '{symbolic!}'
```
Comment (feliam) [inline manticore/platforms/evm.py]: For `data` it provides a concrete solution but for `value` `{symbolic!}`.
Maybe provide a concrete solution for value too ?

**Pull request merged**